meson.build

```diff
@@ -0,0 +1,822 @@
+    else
+        error('Failed to compile and run check-deps/__cxa_demangle.c')
+    endif
+else
+    message('Disabling libstdc++ (disabled by: with-libstdcxx)')
+endif
+
+############################
+# -Dwith-perf build option #
+############################
+
+perf_clockid_is_enabled = false
+have_perf_clockid_code = run_command('cat', files('check-deps/__perf_clockid.c')).stdout().strip()
+have_perf_clockid_result = cc.run(have_perf_clockid_code, name: 'perf clockid check')
+if get_option('with-perf').auto()
+    if have_perf_clockid_result.returncode() == 0
+        perf_clockid_is_enabled = true
+        common_cflags += '-DHAVE_PERF_CLOCKID'
+        message('Enabling perf event support')
+    else
+        message('Disabling perf event support')
+    endif
+elif get_option('with-perf').enabled()
+    if have_perf_clockid_result.returncode() == 0
+        perf_clockid_is_enabled = true
+        common_cflags += '-DHAVE_PERF_CLOCKID'
+        message('Enabling perf event support (enabled by: with-perf)')
+    else
+        error('Failed to compile and run check-deps/__perf_clockid.c')
+    endif
+else
+    message('Disabling perf event support (disabled by: with-perf)')
+endif
+
+################################
+# -Dwith-schedule build option #
+################################
+
+perf_ctxsw_is_enabled = false
+have_perf_ctxsw_code = run_command('cat', files('check-deps/__perf_context_switch.c')).stdout().strip()
+have_perf_ctxsw_result = cc.run(have_perf_ctxsw_code, name: 'perf context switch check')
+if get_option('with-schedule').auto()
+    if have_perf_ctxsw_result.returncode() == 0
+        perf_ctxsw_is_enabled = true
+        common_cflags += '-DHAVE_PERF_CTXSW'
+        message('Enabling scheduler event support')
+    else
+        message('Disabling scheduler event support')
+    endif
+elif get_option('with-schedule').enabled()
+    if have_perf_clockid_result.returncode() == 0
+        perf_clockid_is_enabled = true
+        common_cflags += '-DHAVE_PERF_CLOCKID'
+        message('Enabling scheduler event support (enabled by: with-schedule)')
+    else
+        error('Failed to compile and run check-deps/__perf_context_switch.c')
+    endif
+else
+    message('Disabling scheduler event support (disabled by: with-schedule)')
+endif
+
+#########################
+# required dependencies #
+#########################
+
+libpthread_dep = cc.find_library('pthread', required: true)
+libdl_dep = cc.find_library('dl', required: true)
+librt_dep = cc.find_library('rt', required: true)
+
+###########
+# version #
+###########
+
+version_sh = find_program('misc/version.sh')
+version_sh_cmd = '$(git describe --tags --abbrev=4 --match="v[0-9].[0-9]*" 2> /dev/null || echo v' + meson.project_version() + ')'
+version_h = custom_target(
+    'gen_version_h',
+    output: 'version.h',
+    command: [program_shell, '-c', version_sh.path() + ' @OUTPUT@' + ' ' + version_sh_cmd + ' @0@ .'.format(arch) + ' > /dev/null'],
```

Comment:
We only put arch info to version string, but we have to add other feature info as well.

Reply:
The version string generation is done at https://github.com/namhyung/uftrace/blob/master/misc/version.sh#L33-L55.